test.py

```diff
@@ -107,10 +107,10 @@ def download_test(self, filename, recursive, destination=None,
         os.mkdir(temp_in)
         os.chdir(temp_in)
         try:
-            with SCPClient(self.ssh.get_transport()) as scp:
-                scp.get(filename,
-                        destination if destination is not None else u'.',
-                        preserve_times=True, recursive=recursive)
+            get(self.ssh.get_transport(),
```

Comment:
Oh, I didn't see `test_up_and_down`.

Yeah, a straight duplication sounds bad. You could try a variation on the Abstract Test Case:
http://c2.com/cgi/wiki?AbstractTestCases

to run the same test suite over the different varieties, but it would be more complicated.

If you're happy with coverage, so am I :-)